pennylane_honeywell/device.py

```diff
@@ -172,9 +172,14 @@ def execute(self, circuit, **kwargs):
 
         self.check_validity(circuit.operations, circuit.observables)
 
-        self._circuit_hash = circuit.hash
+        if tape_mode_active():
+            self._circuit_hash = circuit.graph.hash
+            circuit_str = circuit.graph.to_openqasm()
 
-        circuit_str = circuit.to_openqasm()
+        else:
+
+            self._circuit_hash = circuit.hash
+            circuit_str = circuit.to_openqasm()
```

Comment:
I'm very hesitant to have the plugins dependent on any of our currently tape/non-tape mocking in PennyLane, since this will require a new plugin release once `tape_mode_active()` is removed.

Instead, can we use a EAFP approach here instead?

```python
try:
    self._circuit_hash = circuit.graph.hash
    circuit_str = circuit.graph.to_openqasm()
except AttributeError:
    self._circuit_hash = circuit.hash
    circuit_str = circuit.to_openqasm()
```